python/ClusterLauncher/PBSJob.py

```diff
@@ -47,9 +48,11 @@ def copyFiles(self, job_file):
     # Save current location as PBS_O_WORKDIR
     params['pbs_o_workdir'] = os.getcwd()
 
-    # Copy files (unless they are listed in "no_copy"
+    # Copy files (unless they are listed in "no_copy and no_copy_pattern")
     for file in os.listdir('../'):
-      if os.path.isfile('../' + file) and file != job_file and (not params.isValid('no_copy') or file not in params['no_copy']):
+      if os.path.isfile('../' + file) and file != job_file and \
+         (not params.isValid('no_copy') or file not in params['no_copy']) and \
+         (not params.isValid('no_copy_pattern') or file.find(params['no_copy_pattern']) == -1):
```

Comment:
Well this is technically not a "pattern". You're only supporting substrings here which is fine, but then the parameter name is misleading. I can say `*.cluster` and have this work.

We should probably just make the existing parameter a real pattern instead of making two separate options here.